sdk/test-utils/recorder/src/sanitizer.ts

```diff
@@ -1,230 +1,305 @@
+} = {
+  generalSanitizers: pluralize(
+    makeAddFindReplaceSanitizer("GeneralRegexSanitizer", "GeneralStringSanitizer")
+  ),
+  bodySanitizers: pluralize(
+    makeAddFindReplaceSanitizer("BodyRegexSanitizer", "BodyStringSanitizer")
+  ),
+  headerSanitizers: pluralize(addHeaderSanitizer),
+  uriSanitizers: pluralize(makeAddFindReplaceSanitizer("UriRegexSanitizer", "UriStringSanitizer")),
+  connectionStringSanitizers: pluralize(addConnectionStringSanitizer),
+  bodyKeySanitizers: pluralize(makeAddSanitizer("BodyKeySanitizer")),
+  continuationSanitizers: pluralize(addContinuationSanitizer),
+  removeHeaderSanitizer: addRemoveHeaderSanitizer,
+  oAuthResponseSanitizer: makeAddBodilessSanitizer("OAuthResponseSanitizer"),
+  uriSubscriptionIdSanitizer: makeAddSanitizer("UriSubscriptionIdSanitizer"),
+  resetSanitizer: makeAddBodilessSanitizer("Reset"),
+};
+
+export async function addSanitizers(
+  httpClient: HttpClient,
+  url: string,
+  recordingId: string,
+  options: SanitizerOptions
+): Promise<void> {
+  await Promise.all(
+    Object.entries(options).map(([key, sanitizer]) => {
+      const action = addSanitizersActions[key as keyof SanitizerOptions];
+      if (!action) {
+        throw new RecorderError(`Sanitizer ${key} not implemented`);
       }
-    } else {
-      throw new RecorderError("Bad state, recordingId is not defined when called addSanitizer().");
-    }
+
+      return action(httpClient, url, recordingId, sanitizer);
+    })
+  );
+}
+
+/**
+ * Atomic method to add a simple sanitizer.
+ */
+async function addSanitizer(
+  httpClient: HttpClient,
+  url: string,
+  recordingId: string,
+  options: {
+    sanitizer: ProxyToolSanitizers;
+    body: Record<string, unknown> | undefined;
+  }
+): Promise<void> {
+  const uri = `${url}${paths.admin}${
+    options.sanitizer !== "Reset" ? paths.addSanitizer : paths.reset
+  }`;
+  const req = createRecordingRequest(uri, recordingId);
+  if (options.sanitizer !== "Reset") {
+    req.headers.set("x-abstraction-identifier", options.sanitizer);
   }
+  req.headers.set("Content-Type", "application/json");
+  req.body = options.body !== undefined ? JSON.stringify(options.body) : undefined;
 
-  /**
-   * Adds the recording id headers to the requests that are sent to the proxy tool.
-   * These are required to appropriately save the recordings in the record mode and picking them up in playback.
-   */
-  private _createRecordingRequest(url: string, method: HttpMethods = "POST") {
-    const req = createPipelineRequest({ url: url, method });
-    if (this.recordingId !== undefined) {
-      req.headers.set("x-recording-id", this.recordingId);
+  const rsp = await httpClient.sendRequest({
+    ...req,
+    allowInsecureConnection: true,
+  });
+  if (rsp.status !== 200) {
+    throw new RecorderError("addSanitizer request failed.");
+  }
+}
+
+/**
+ * Returns the html document of all the available transforms in the proxy-tool
+ */
+export async function transformsInfo(
```

Comment:
These functions are all unit tested, right? seems like the perfect opportunity.

Reply:
I don’t see the unit tests 👀 I could be missing something.

Reply:
They are not unit tested at this time :(. I'd love to add them at some point, but perhaps not in this PR -- I'd like to have this in tomorrow in time for the "recorder migration hackathon" that is happening on Friday.

Reply:
All the sanitizers are individually tested with the proxy tool and the fake service/server.
So, should be fine.

"transformsInfo" is not for the public, even we don't use it, to be honest.

It just dumps the documentation of the proxy tool, which Scott wants to host somewhere else later.

Reply:
Unit tests add value in the sense that functionalities can be understood and worked with independently from what is using them, and independently from one another.

I get that we don’t want to add them in this PR.

However, this division in logic puts us in a wonderful position to continue pursuing a more TDD path.

If that sounds interesting, I would recommend adding at least an issue to follow up 🙂

Reply:
Sure 🙂.

Let's log an issue @timovv, please!

Reply:
#20101